src/Component/Dash.js

```diff
@@ -0,0 +1,309 @@
+      duration: theme.transitions.duration.leavingScreen,
+    }),
+    width: theme.spacing(7),
+    [theme.breakpoints.up('sm')]: {
+      width: theme.spacing(9),
+    },
+  },
+  appBarSpacer: theme.mixins.toolbar,
+  content: {
+    flexGrow: 1,
+    height: '100vh',
+    overflow: 'auto',
+  },
+  container: {
+    paddingTop: theme.spacing(4),
+    paddingBottom: theme.spacing(4),
+  },
+  paper: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    overflow: 'auto',
+    flexDirection: 'column',
+  },
+  fixedHeight: {
+    height: 240,
+  },
+}));
+
+function Dash() {
+  function dateFormat(Date) {
+    let dateF = '';
+    const year = Date.getFullYear();
+    const month = Date.getMonth() + 1;
+    const day = Date.getUTCDate();
+    dateF = `${year}-${month}-${day}`;
+    return dateF;
+  }
+
+  const classes = useStyles();
+  const navigate = useNavigate();
+  const logout = () => {
+    localStorage.clear();
+    navigate('/');
+  };
+
+  const [openDrawer, setOpenDrawer] = useState(false);
+  const handleDrawerOpen = () => {
+    setOpenDrawer(true);
+  };
+  const handleDrawerClose = () => {
+    setOpenDrawer(false);
+  };
+
+  const [isAddLeaves, setAddLeaves] = useState(false);
+  const [pLeave, setPLeave] = useState(false);
+  const [date, setDate] = useState('');
+
+  const openDiag = () => {
+    setAddLeaves(true);
+  };
+
+  const closeDiag = () => {
+    setAddLeaves(false);
+  };
+
+  const handleSubmit = () => {
+    setPLeave(true);
+  };
+
+  // declare a new state variable for modal open
+  const [open, setOpen] = useState(false);
+
+  // function to handle modal open
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  // function to handle modal close
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <div className={classes.root}>
+      <CssBaseline />
+      <AppBar
+        style={{ background: '#15317E' }}
+        position="absolute"
+        className={clsx(classes.appBar, openDrawer && classes.appBarShift)}
+      >
+        <Toolbar className={classes.toolbar}>
+          <IconButton
+            edge="start"
+            color="inherit"
+            aria-label="open drawer"
+            onClick={handleDrawerOpen}
+            className={clsx(
+              classes.menuButton,
+              openDrawer && classes.menuButtonHidden,
+            )}
+          >
+            <MenuIcon />
+          </IconButton>
+          <Typography
+            component="h1"
+            variant="h6"
+            color="inherit"
+            noWrap
+            className={classes.title}
+          >
+            Leave Tracker
+          </Typography>
+          <Typography
+            component="h3"
+            variant="h6"
+            color="inherit"
+            noWrap
+            className={classes.title}
+          >
+            Hi,{' '}
+            {localStorage.getItem('first_name') +
+              localStorage.getItem('last_name')}
```

Comment:
Wouldn't this be done at every re-render? This is an expensive operation, should we do it in state?